eth-bytecode-db/eth-bytecode-db/src/search/full_match.rs

```diff
@@ -0,0 +1,59 @@
+use super::{BytecodeRemote, MatchContract};
+use crate::verification::MatchType;
+use entity::{sea_orm_active_enums::BytecodeType, sources};
+use sea_orm::{entity::prelude::*, ConnectionTrait, FromQueryResult, Statement};
+
+pub async fn find_full_match_contract<C>(
+    db: &C,
+    remote: &BytecodeRemote,
+) -> Result<Vec<MatchContract>, anyhow::Error>
+where
+    C: ConnectionTrait,
+{
+    let candidates = find_source_candidates(db, remote).await?;
+    let mut matches = vec![];
+    for candidate in candidates.into_iter() {
```

Comment:
In general, in case of "full" match only one source code must be valid. If that is not true, then it may indicate some error in our program execution (most probably, that we incorrectly filling the database on verification).

I suggest to add a `tracing::error` in case if `candidates` contains more than one item, so that we would log such situations and can check them manually.

Reply:
good point